### PR TITLE
ci: Several glibc related fixes

### PR DESCRIPTION
**Pull request by Dennisbonke**
This should fix the ansi only build on the ci.

### CODE REVIEW

File: options/ansi/musl-generic-math/libm.h [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #include <float.h>
 #include <math.h>
 //#include <complex.h>
-#include <endian.h>
+// #include <endian.h>
```
Comment (ArsenArsen) [inline options/ansi/musl-generic-math/libm.h:20]: why was this done? why is it a comment

Comment (Dennisbonke) [reply options/ansi/musl-generic-math/libm.h:20]: endian.h is provided by the glibc option. I followed the style of complex.h and just commented it out.

Comment (ArsenArsen) [reply options/ansi/musl-generic-math/libm.h:20]: why isn't it included with it if it is? is it even required or just a false transitive include?

Comment (Dennisbonke) [reply options/ansi/musl-generic-math/libm.h:20]: We can't use that if we build ansi-only?

Comment (Geertiebear) [inline options/ansi/musl-generic-math/libm.h:20]: Don't comment it out, just remove it. Same for the ``complex.h`` include.

**Review by Geertiebear — CHANGES_REQUESTED** [review]

File: options/ansi/musl-generic-math/libm.h (hunk 1 above)
Comment (Geertiebear) [reply options/ansi/musl-generic-math/libm.h:20]: This is not a public header, and the CI passes without the ``endian.h`` include, so we can assume nothing will break with this change.

Comment (Dennisbonke) [reply options/ansi/musl-generic-math/libm.h]: Done as requested, should be fine now pending CI completion.

**Review by Geertiebear — APPROVED** [review]
LGTM now.

**Pull request merged**